File: cpp/src/Comms/RabbitMQ/RabbitPublisher.cpp
//
// Created by Peter on 6/24/2024.
//

#include "RabbitPublisher.h"

#include <stdexcept>

void RabbitPublisher::publishMessage(const std::string &message, bool persistent) {
    if (!channel) {
        throw std::runtime_error("Not connected to RabbitMQ. Call connect() first.");
    }

    AmqpClient::BasicMessage::ptr_t msg = AmqpClient::BasicMessage::Create(message);
    if (!persistent) {
        msg->DeliveryMode(AmqpClient::BasicMessage::dm_nonpersistent);
    }

    channel->BasicPublish(exchange, routing_key, msg);
}

File: python/WavesDisplay/comms/rabbit_publisher.py
import pika
from comms.rabbit_client import RabbitClient


class RabbitPublisher(RabbitClient):
    def publish_message(self, message: str, persistent: bool = False):
        if self.channel is None:
            raise Exception("Not connected to RabbitMQ. Call connect() first.")

        properties = None
        if not persistent:
            properties = pika.BasicProperties(delivery_mode=1)
        self.channel.basic_publish(exchange=self.exchange, routing_key=self.routing_key, body=message)
